src/docs/arch-checkpoint.dox

```diff
@@ -0,0 +1,62 @@
+/*! @arch_page arch-checkpoint Checkpoint
+
+# Overview #
+
+A checkpoint is a known point in time from which WiredTiger can recover in the event of a
+crash or unexpected shutdown. WiredTiger checkpoints are created either via the API
+WT_SESSION::checkpoint, or internally. Internally checkpoints are created on startup, shutdown
+and during compaction.
+
+A checkpoint is performed within the context of snapshot isolation transaction as such the
+checkpoint has a consistent view of the database from beginning to end. The checkpoint can be run
+with or without a timestamp, generally a timestamp is used. This timestamp is not used as a
+\c read_timestamp for the checkpoint transaction. The timestamp provided is set to the checkpoint
```

Comment:
Can we talk about `stable timestamp` here?

Reply:
Sure I'll add in some detail about that.